scripts/man-dates.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+# This script updates the date lines in the man pages to the date of the last
+# commit to that file.
+
+set -eu
+
+for i in $(find man -type f) ; do
```

Comment:
It looks like shellcheck [identified](http://build.zfsonlinux.org/builders/Ubuntu%2017.10%20x86_64%20%28STYLE%29/builds/13227/steps/shell_2/logs/stdio) some concerns with the script.  Let's either suppress the warnings or resolve them as suggested.

```
./scripts/man-dates.sh:8:10: warning: For loops over find output are fragile. Use find -exec or a while read loop. [SC2044]
./scripts/man-dates.sh:9:66: warning: This only exits the subshell caused by the $(..) expansion. [SC2106]
Makefile:1595: recipe for target 'shellcheck' failed
```

Reply:
Fixed.